modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/services/Addresses.scala

```diff
@@ -61,5 +63,29 @@ object Addresses {
         unconfirmed = Balance(offChainBalance, offChainAssets.map(TokenAmount(_)))
       )) ||> trans.xa
     }
+
+    private def hasBeenUsedByErgoTree(ergoTree: HexString): F[Boolean] =
+      outputRepo.nodeOutputCount(ergoTree).map(_ > 0) ||> trans.xa
+
+    private def addressInfoOf(
+      address: Address
+    ): F[(Address, AddressInfo)] = {
+      val tree = sigma.addressToErgoTreeNewtype(address)
+      for {
+        balance <- confirmedBalanceOf(address, 0)
+        hUTxs   <- memprops.hasUnconfirmedBalance(tree)
+        used    <- hasBeenUsedByErgoTree(tree.value)
+      } yield (address, AddressInfo(address = address, hasUnconfirmedTxs = hUTxs, used = used, balance))
+    }
+
+    def addressInfoOf(batch: List[Address]): F[Map[Address, AddressInfo]] =
+      batch.distinct
+        .map(addressInfoOf)
```

Comment:
This approach is totally inefficient. Why not using batch request (e.g. `where address in (..,..)`)?